xmake/rules/c++/modules/modules_support/builder.lua

```diff
@@ -460,3 +461,32 @@ function add_headerunit_to_target_mapper(target, headerunit, bmifile)
     return deduplicated and true or false
 end
 
+-- check if dependencies changed
+function is_dependencies_changed(target, module)
+    local cachekey = target:name() .. module.name
+
+    local requires
+    for required, _ in table.orderpairs(module.requires) do
+        requires = requires or hashset.new()
+        requires:insert(required)
+    end
+
+    local oldrequires = compiler_support.memcache():get2(cachekey, "oldrequires")
```

Comment:
if we use memcache, why not save oldrequires as hashset instead of array? it will be faster.